src/melee.cpp

```diff
@@ -151,7 +151,7 @@ int player::get_hit_weapon( const item &weap ) const
     ///\EFFECT_BASHING improves hit chance for bashing weapons
     ///\EFFECT_CUTTING improves hit chance for cutting weapons
     ///\EFFECT_STABBING improves hit chance for piercing weapons
-    auto bonus = get_skill_level( weap.melee_skill() );
+    auto bonus = get_skill_level( is_armed() ? weap.melee_skill() : skill_unarmed );
```

Comment:
Null item should have null skill. I'm not sure having it lie about that is a good idea - it starts out harmless but as the code evolves we may come to regret that